### DISCUSSION TITLE
Deploying additional nodes after initial deployment

### DISCUSSION THREAD

**Discussion by Venoox** [onedr0p/cluster-template#589] [Q&A] [answered]
If I want to deploy more nodes, do I just need to add them to the config and run the whole process again?

**Comment by onedr0p** [comment, marked as answer]
I converted your issue to a discussion.

Yes that's pretty much it but note:
- Going from 1 to 2 master nodes is not supported, you are on your own to set up an external postgres database and migrate.
- Going from 1 to 3 master nodes is supported as long as the existing master is using etcd.

With that out of the way, add more nodes, should be as easy as adding the nodes to your Ansible `hosts.yml` in the correct group you want and then doing the following:

> It's always a good to have application data backed up before continuing in case something goes wrong.

1. Verify the new nodes are in the config:`task ansible:list`
2. Verify you can ping them: `task ansible:ping`
3. Add the OS bits to the nodes: `task ansible:prepare`
4. Verify the k3s version in your [k3s configuration](https://github.com/onedr0p/flux-cluster-template/blob/2ae3bc7cba88ffdf7e6b70e9535aea44fb3d90cd/ansible/inventory/group_vars/kubernetes/k3s.yml#L9) matches your cluster version and run `task ansible:install`
